platform/database/tests/database_test.go

```diff
@@ -466,39 +468,48 @@ func TestProdDatabase(t *testing.T) {
 			{
 				name: "when a condition is provided it only gets the requested data",
 				condition: &struct {
-					sql string
-					arg []string
-				}{sql: "WHERE id IN (?)", arg: []string{firstData.ID}},
+					sql  string
+					args []interface{}
+				}{sql: "WHERE id IN (?)", args: []interface{}{firstData.ID}},
 				shouldFindData: []testData{firstData},
 				shouldErr:      false,
 			},
 			{
 				name: "when multiple conditions are given it gets all the request data",
 				condition: &struct {
-					sql string
-					arg []string
-				}{sql: "WHERE id IN (?)", arg: []string{firstData.ID, secondData.ID}},
+					sql  string
+					args []interface{}
+				}{sql: "WHERE id IN (?)", args: []interface{}{[]string{firstData.ID, secondData.ID}}},
 				shouldFindData: []testData{firstData, secondData},
 				shouldErr:      false,
 			},
 			{
 				name: "when no data is matching the condition, it returns an empty slice with no error",
 				condition: &struct {
-					sql string
-					arg []string
-				}{sql: "WHERE id IN (?)", arg: []string{"3237b466-b3c6-4521-96c5-61022c4a1796"}},
+					sql  string
+					args []interface{}
+				}{sql: "WHERE id IN (?)", args: []interface{}{"3237b466-b3c6-4521-96c5-61022c4a1796"}},
 				shouldFindData: []testData{},
 				shouldErr:      false,
 			},
 			{
 				name: "when the condition is faulty, it does not populate the slice and return an error",
 				condition: &struct {
-					sql string
-					arg []string
-				}{sql: "WHERE non_exisiting_field IN (?) ", arg: []string{firstData.ID}},
+					sql  string
+					args []interface{}
+				}{sql: "WHERE non_exisiting_field IN (?) ", args: []interface{}{firstData.ID}},
 				shouldFindData: []testData{},
 				shouldErr:      true,
 			},
+			{
+				name: "when a condition is provided with arguments of different types, it returns the expected data",
+				condition: &struct {
+					sql  string
+					args []interface{}
+				}{sql: "WHERE id IN (?) AND number IN (?)", args: []interface{}{firstData.ID, firstData.Number}},
+				shouldFindData: []testData{firstData},
+				shouldErr:      false,
```

Comment:
this is an interesting test but what happens if both are lists ? 
How do you do `WHERE id IN [2, 3, 4] AND Number IN [7, 8, 9]` ?

Reply:
> How do you do WHERE id IN [2, 3, 4] AND Number IN [7, 8, 9] ?

`args: []interface{}{[]string{"2", "3", "4"}, []int{2,3,4}}}`

I can add another test for this :)

Reply:
ba actually your testing IN on single values, which is not really likely to happen, is it ?

Reply:
Well the query I need the change for is of this form:

`WHERE id = (?) AND value IN (?)`

And in this case we'd pass a single value, then a slice of values in the `[]interface{}` , so having a test with single values is not that bad.

Reply:
but you are testing with `IN` here not `=`

Reply:
and I think both tests are valuable, I just think testing that IN maps a non array value is not as important as testing it works with several arrays or a mix of arrays and non arrays.

Reply:
Sure I'll add the interesting cases as well.

Also fun little bit about the `IN` vs `=`, `sqlx.In()` does not check if there's an IN at all if I understood the code well. It just checks for the `?`s and proceeds to set the right list of arguments in those places.